sentence_transformers/sparse_encoder/model_card.py

```diff
@@ -0,0 +1,152 @@
+
+
+class SparseEncoderModelCardCallback(SentenceTransformerModelCardCallback):
+    def __init__(self, default_args_dict: dict[str, Any]) -> None:
+        super().__init__(default_args_dict)
+
+
+@dataclass
+class SparseEncoderModelCardData(SentenceTransformerModelCardData):
+    """A dataclass storing data used in the model card.
+
+    Args:
+        language (`Optional[Union[str, List[str]]]`): The model language, either a string or a list,
+            e.g. "en" or ["en", "de", "nl"]
+        license (`Optional[str]`): The license of the model, e.g. "apache-2.0", "mit",
+            or "cc-by-nc-sa-4.0"
+        model_name (`Optional[str]`): The pretty name of the model, e.g. "SparseEncoder based on answerdotai/ModernBERT-base".
+        model_id (`Optional[str]`): The model ID when pushing the model to the Hub,
+            e.g. "tomaarsen/se-mpnet-base-ms-marco".
+        train_datasets (`List[Dict[str, str]]`): A list of the names and/or Hugging Face dataset IDs of the training datasets.
+            e.g. [{"name": "SNLI", "id": "stanfordnlp/snli"}, {"name": "MultiNLI", "id": "nyu-mll/multi_nli"}, {"name": "STSB"}]
+        eval_datasets (`List[Dict[str, str]]`): A list of the names and/or Hugging Face dataset IDs of the evaluation datasets.
+            e.g. [{"name": "SNLI", "id": "stanfordnlp/snli"}, {"id": "mteb/stsbenchmark-sts"}]
+        task_name (`str`): The human-readable task the model is trained on,
+            e.g. "semantic search and sparse retrieval".
+        tags (`Optional[List[str]]`): A list of tags for the model,
+            e.g. ["sentence-transformers", "sparse-encoder"].
+
+    .. tip::
+
+        Install `codecarbon <https://github.com/mlco2/codecarbon>`_ to automatically track carbon emission usage and
+        include it in your model cards.
+
+    Example::
+
+        >>> model = SparseEncoder(
+        ...     "microsoft/mpnet-base",
+        ...     model_card_data=SparseEncoderModelCardData(
+        ...         model_id="tomaarsen/se-mpnet-base-allnli",
+        ...         train_datasets=[{"name": "SNLI", "id": "stanfordnlp/snli"}, {"name": "MultiNLI", "id": "nyu-mll/multi_nli"}],
+        ...         eval_datasets=[{"name": "SNLI", "id": "stanfordnlp/snli"}, {"name": "MultiNLI", "id": "nyu-mll/multi_nli"}],
+        ...         license="apache-2.0",
+        ...         language="en",
+        ...     ),
+        ... )
+    """
+
+    # Potentially provided by the user
+    task_name: str = field(default=None)
+    tags: list[str] | None = field(
+        default_factory=lambda: [
+            "sentence-transformers",
+            "sparse-encoder",
+        ]
+    )
+
+    # Automatically filled by `SparseEncoderModelCardCallback` and the Trainer directly
+    predict_example: list[list[str]] | None = field(default=None, init=False)
+
+    # Computed once, always unchanged
+    pipeline_tag: str = field(default=None, init=False)
```

Comment:
```suggestion
    pipeline_tag: str = field(default=None, init=False)
    template_path: Path = field(default=Path(__file__).parent / "model_card_template.md", init=False)
```

The `SentenceTransformer` class now uses `model.model_card_data.template_path` as the template path right now. Adding this line should make the model card generation work with your updated template.